### PR TITLE
support docker and cis version

### PR DESCRIPTION
**Pull request by yoavrotems**
support both use of docker version and benchmark version input https://github.com/aquasecurity/docker-bench/issues/83

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -26,14 +26,23 @@ func app(cmd *cobra.Command, args []string) {
 	var err error
 
 	// Get version of Docker benchmark to run
-	if dockerVersion != "" {
-		version = dockerVersion
+	if benchmarkVersion != "" {
+		version = benchmarkVersion
```
Comment (lizrice) [inline app.go:30]: I think you also need to test whether `dockerVersion` is empty string - if the user specifies both dockerVersion and benchmarkVersion it needs to return an error

**Review by lizrice — CHANGES_REQUESTED** [review]
Missing an error case I think? But otherwise looks good

File: app.go (hunk 1 above)
Comment (yoavrotems) [reply app.go:30]: Do we need to return error? I think that currently if the user specifies both then the benchmark will be the one to take effect, maybe we should print the error for logging? (but still run it)

File: app.go [hunk 2]
```diff
@@ -26,14 +26,27 @@ func app(cmd *cobra.Command, args []string) {
...
+				fmt.Errorf("Version check failed: Error can't specify both --version and --benchmark flags\nIf not specify neither docker version will be auto detect"))
+		}
+		version = benchmarkVersion
+	} else {
+		if dockerVersion != "" {
+			version, err = getDockerCisVersion(dockerVersion)
+			if err != nil {
+				util.ExitWithError(
+					fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
+						err))
```
Comment (lizrice) [inline app.go:41]: This error message is misleading - in this case the user has supplied a version with `--version` but it couldn't be converted to a CIS benchmark version

```suggestion
				fmt.Errorf("Failed to get a valid CIS benchmark version for Docker version %s: %v",
					dockerVersion, err))
```

File: app.go [hunk 3]
```diff
@@ -26,14 +26,27 @@ func app(cmd *cobra.Command, args []string) {
...
+			}
+		} else {
+			version, err = getDockerVersion()
+			if err != nil {
+				util.ExitWithError(
+					fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
+						err))
+			}
```
Comment (lizrice) [inline app.go:49]: What you have will work but I think it could be neater. Whether the user suggests a Docker version (with `--version`), or whether it gets auto-detected, we need to convert that into a CIS benchmark with the `getDockerCisVersion()` function. I would call that function just once, from here, and not also call it from `getDockerVersion()`. So the logic would be like this:   

```
dockerVersion is the user-supplied docker version
if dockerVersion is empty:
  dockerVersion = auto-detected dockerVersion
benchmarkVersion = convert dockerVersion to benchmark version
```

The changed code would be something like this: 
```suggestion
		if dockerVersion == "" {
		        // Auto-detect the version of Docker that's running
			dockerVersion, err = getDockerVersion()
			if err != nil {
				util.ExitWithError(
					fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
						err))
		}
		
		// Find the CIS benchmark version for this version of Docker
		version, err = getDockerCisVersion(dockerVersion)
		if err != nil {
			util.ExitWithError(
				fmt.Errorf("Failed to get a valid CIS benchmark version for Docker version %s: %v",
					dockerVersion, err))
		}
```

Note that you'd have to remove the call to `getDockerCisVersion()` at 139. But also note that as it currently stands, the comment for `getDockerVersion()` at L132 is incorrect - it says it returns the docker server engine version but if you leave it calling `getDockerCisVersion()` then it's really returning the CIS benchmark version.

**Review by lizrice — CHANGES_REQUESTED** [review]
I think we can make this better :-) give me a shout if you think I am being too picky or if you want to talk it through

File: app.go (hunk 1 above)
Comment (lizrice) [reply app.go:38]: We should be consistent with kube-bench, and that errors if you try to specify both. See the help text at https://github.com/aquasecurity/kube-bench/blob/58bea9c89b9be43ebdf76de256d916a608df43c4/cmd/root.go#L176 (and also the help text you've used in root.go below!)

File: app.go [hunk 4]
```diff
@@ -25,15 +25,33 @@ func app(cmd *cobra.Command, args []string) {
...
-	// Get version of Docker benchmark to run
-	if dockerVersion != "" {
-		version = dockerVersion
+	// Benchmark flag is specify
+	if benchmarkVersion != "" {
+
+		// Check for not specify both --version and --benchmark
+		if dockerVersion != "" {
+			util.ExitWithError(
+				fmt.Errorf("Version check failed: Error can't specify both --version and --benchmark flags\nIf not specify neither docker version will be auto detect"))
```
Comment (lizrice) [inline app.go:34]: See https://github.com/aquasecurity/kube-bench/blob/714430c7fcb718de0b9d9d350aec26445d218b46/cmd/common.go#L279

```suggestion
				fmt.Errorf("It is an error to specify both --version and --benchmark flags"))
```

**Review by lizrice — CHANGES_REQUESTED** [review]

**Review by danielpacak — APPROVED** [review]
LGTM!

You can think of `ResolveCisVersion(benchmarkVersion, dockerVersion string) (string,error)` method that will wrap all these if/else statements and will allow you to unit test this logic.

**Review by lizrice — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -14,12 +14,12 @@ docker-bench currently supports tests as defined in the following CIS Docker Ben
...
-| [Docker Benchmark v1.2.0](https://workbench.cisecurity.org/benchmarks/2351) | 18.09 | 18.09 and Docker Enterprise 2.1 |
-| [Docker Community Edition Benchmark v1.1.0](https://workbench.cisecurity.org/benchmarks/552) | 17.06 | 17.06 |
-| [Docker Benchmark v1.0.0](https://workbench.cisecurity.org/benchmarks/363) | 1.13.0 | 1.13.0 |
+| [Docker Benchmark v1.2.0](https://workbench.cisecurity.org/benchmarks/2351) | cis-1.2 | 18.09 and Docker Enterprise 2.1 |
+| [Docker Community Edition Benchmark v1.1.0](https://workbench.cisecurity.org/benchmarks/552) | cis-1.1 | 17.06 |
+| [Docker Benchmark v1.0.0](https://workbench.cisecurity.org/benchmarks/363) | cis-1.0 | 1.13.0 |
 
 docker-bench will determine the test set to run based on the Docker version running on the host machine. 
-The version to run tests for can also be specified manually with the `--version <ver>` commandline flag.
+The version to run tests for can also be specified manually with the `--version <ver>` or `--benchmark <var>` commandline flag.
```
Comment (lizrice) [inline README.md:22]:
```suggestion
The version to run tests for can also be specified manually with the `--version <Docker version>` or `--benchmark <CIS benchmark version>` commandline flag.
```

**Review by lizrice — APPROVED** [review]
